File: src/dataset/c/pangram_exercism-viettronics868/openAI/ChatGPT4_pangram_v4.c
#include "pangram.h"
#include <stdbool.h>
#include <ctype.h>

bool is_pangram(const char *sentence) {
    if (!sentence) return false;

    bool seen[26] = {false};
    int unique_count = 0;

    for (const char *ptr = sentence; *ptr; ++ptr) {
        if (isalpha((unsigned char)*ptr)) {
            int index = tolower((unsigned char)*ptr) - 'a';
            if (!seen[index]) {
                seen[index] = true;
                unique_count++;
                if (unique_count == 26) return true;
            }
        }
    }

    return false;
}

File: src/dataset/typescript/pangram_exercism-wackerow/openAI/ChatGPT4_pangram_v4.ts
export default class {
  constructor(private sentence: string) {}

  isPangram(): boolean {
    const seen = new Set<string>();
    for (const char of this.sentence.toLowerCase()) {
      if (char >= 'a' && char <= 'z') {
        seen.add(char);
        if (seen.size === 26) return true;
      }
    }
    return false;
  }
}